File: BigDigits/bd.cpp
// big digits cpp soln for code eval by steven a dunn

#include <cctype>
#include <fstream>
#include <iostream>
#include <string>
#include <vector>

using std::cout;
using std::endl;
using std::ifstream;
using std::string;
using std::vector;

void printBigDigits(string);
void bigPrint(char, vector<string>&);

int main (int argc, char* argv[])
{
    ifstream inputFile(argv[1]);
    string line;

    if (inputFile)
    {
        while (getline(inputFile, line))
            printBigDigits(line);
        inputFile.close();
    }

    return 0;
}

void printBigDigits(string line)
{
    vector<string> rows(5);
    for (int i = 0; i < rows.size(); ++i)
        rows[i] = "";

    for (int i = 0; i < line.size(); ++i)
        if (isdigit(line[i]))
            bigPrint(line[i], rows);

    for (int i = 0; i < rows.size(); ++i)
        cout << rows[i] << endl;

    string spacer = "";
    for (int i = 0; i < rows[0].size(); ++i)
        spacer += "-";
    cout << spacer << endl;
}

void bigPrint(char c, vector<string>& rows)
{
    int val = c - '0';

    if (val == 1)
    {
        rows[0] += "--*--";
        rows[1] += "-**--";
        rows[2] += "--*--";
        rows[3] += "--*--";
        rows[4] += "-***-";
    }
    else if (val == 2)
    {
        rows[0] += "***--";
        rows[1] += "---*-";
        rows[2] += "-**--";
        rows[3] += "*----";
        rows[4] += "****-";
    }
    else if (val == 3)
    {
        rows[0] += "***--";
        rows[1] += "---*-";
        rows[2] += "-**--";
        rows[3] += "---*-";
        rows[4] += "***--";
    }
    else if (val == 4)
    {
        rows[0] += "-*---";
        rows[1] += "*--*-";
        rows[2] += "****-";
        rows[3] += "---*-";
        rows[4] += "---*-";
    }
    else if (val == 5)
    {
        rows[0] += "****-";
        rows[1] += "*----";
        rows[2] += "***--";
        rows[3] += "---*-";
        rows[4] += "***--";
    }
    else if (val == 6)
    {
        rows[0] += "-**--";
        rows[1] += "*----";
        rows[2] += "***--";
        rows[3] += "*--*-";
        rows[4] += "-**--";
    }
    else if (val == 7)
    {
        rows[0] += "****-";
        rows[1] += "---*-";
        rows[2] += "--*--";
        rows[3] += "-*---";
        rows[4] += "-*---";
    }
    else if (val == 8)
    {
        rows[0] += "-**--";
        rows[1] += "*--*-";
        rows[2] += "-**--";
        rows[3] += "*--*-";
        rows[4] += "-**--";
    }
    else if (val == 9)
    {
        rows[0] += "-**--";
        rows[1] += "*--*-";
        rows[2] += "-***-";
        rows[3] += "---*-";
        rows[4] += "-**--";
    }
    else if (val == 0)
    {
        rows[0] += "-**--";
        rows[1] += "*--*-";
        rows[2] += "*--*-";
        rows[3] += "*--*-";
        rows[4] += "-**--";
    }
}

File: BigDigits/py2/bd.py
# big digits py2 soln for code eval by steven a dunn

import sys

def printBigDigits(line):
    rows = ["", "", "", "", ""]
    for i in range(0, len(line)):
        if line[i].isdigit():
            bigPrint(line[i], rows)
    for row in rows:
        print row
    spacer = "-" * len(rows[0])
    print spacer

def bigPrint(char, rows):
    char = int(char)
    if char == 1:
        rows[0] += "--*--";
        rows[1] += "-**--";
        rows[2] += "--*--";
        rows[3] += "--*--";
        rows[4] += "-***-";
    elif char == 2:
        rows[0] += "***--";
        rows[1] += "---*-";
        rows[2] += "-**--";
        rows[3] += "*----";
        rows[4] += "****-";
    elif char == 3:
        rows[0] += "***--";
        rows[1] += "---*-";
        rows[2] += "-**--";
        rows[3] += "---*-";
        rows[4] += "***--";
    elif char == 4:
        rows[0] += "-*---";
        rows[1] += "*--*-";
        rows[2] += "****-";
        rows[3] += "---*-";
        rows[4] += "---*-";
    elif char == 5:
        rows[0] += "****-";
        rows[1] += "*----";
        rows[2] += "***--";
        rows[3] += "---*-";
        rows[4] += "***--";
    elif char == 6:
        rows[0] += "-**--";
        rows[1] += "*----";
        rows[2] += "***--";
        rows[3] += "*--*-";
        rows[4] += "-**--";
    elif char == 7:
        rows[0] += "****-";
        rows[1] += "---*-";
        rows[2] += "--*--";
        rows[3] += "-*---";
        rows[4] += "-*---";
    elif char == 8:
        rows[0] += "-**--";
        rows[1] += "*--*-";
        rows[2] += "-**--";
        rows[3] += "*--*-";
        rows[4] += "-**--";
    elif char == 9:
        rows[0] += "-**--";
        rows[1] += "*--*-";
        rows[2] += "-***-";
        rows[3] += "---*-";
        rows[4] += "-**--";
    elif char == 0:
        rows[0] += "-**--";
        rows[1] += "*--*-";
        rows[2] += "*--*-";
        rows[3] += "*--*-";
        rows[4] += "-**--";
    else:
        pass
    return rows

f = open(sys.argv[1], 'r')
for line in f:
    printBigDigits(line)
f.close()